app/models/concerns/order_workflow.rb

```diff
@@ -19,16 +19,18 @@ module OrderWorkflow
         event :cancel, transitions_to: :cancelled, if: ->(order) { order.check_if_admin }
       end
       state :completed do
-        event :deliver, transitions_to: :delivered, if: ->(order) { order.check_if_admin }
-        event :cancel, transitions_to: :cancelled, if: ->(order) { order.check_if_admin }
+        event :deliver, transitions_to: :delivered
+        event :cancel, transitions_to: :cancelled
       end
       state :delivered
       state :cancelled
 
       before_transition do |_from_state, to_state, _event|
-        check_deals_availability if DEAL_AVAILABILITY_STATES.include?(to_state)
-        check_deals_expiry
-        check_deals_publishability
+        if to_state != :cancelled
```

Comment:
we can also write it as return if to_state != :cancelled
so that we don't have to wrap the entire code